components/scream/src/physics/p3/atmosphere_microphysics.cpp

```diff
@@ -32,52 +43,102 @@ void P3Microphysics::set_grids(const std::shared_ptr<const GridsManager> grids_m
   // Nevertheless, for output reasons, we like to see 'kg/kg'.
   auto Q = kg/kg;
   Q.set_string("kg/kg");
-
-  constexpr int NVL = SCREAM_NUM_VERTICAL_LEV;
-  constexpr int QSZ =  35;  /* TODO THIS NEEDS TO BE CHANGED TO A CONFIGURABLE */
+  auto nondim = m/m;
```

Comment:
Maybe a silly question, but is there a "non-dimensional" unit supported by EKAT? "Meters per meter" sounds non-dimensional until I think about all the "kg / kg" and "mol / mol" stuff I've been seeing. @bartgol ?

Reply:
not too silly.  This is kindof a hack, and maybe @bartgol  can comment on whether or not there is a legit nondim defined in EKAT.  kg/kg is a bit of a special unit because it represents kg/kg_air so it is "non-dimensional" but not unitless.  If that makes sense.

Reply:
It does. I'm actually in favor of more carefully annotating these quantities, at least in our documentation, to avoid confusing non-experts. In EAGLES we deal with various of those ratios involving "air", aerosols, and gases.

Reply:
I thought I added a `ekat::units::nondim` declaration, but I guess I didn't. Doing `auto nondim = kg/kg` obviously work, and does the same thing. The mixing ratio is indeed unitless, except that we hack its string so that it prints `kg/kg`, by doing something like `auto q = kg/kg;  q.set_string("kg/kg");`. However, internally, `q` and `nondim` store the same "dimensions".

That said, we can declare a `nondim` unit in ekat. Or you could create one less ambiguously like so `ekat::Units nondim(0,0,0,0,0,0,0);`.

Reply:
I think actually might have:
```  static Units nondimensional () {
    return Units(RationalConstant::one());
  }```